go/gilded-rose_test.go

```diff
@@ -1,15 +1,108 @@
 package main
 
-import "testing"
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
 
-func Test_Foo(t *testing.T) {
-	var items = []*Item{
-		&Item{"foo", 0, 0},
+func TestUpdateQuality(t *testing.T) {
+	var tests = []struct {
+		testname string
+		in       []*Item
+		want     []*Item
+	}{
+		{testname: "A 1 1", in: []*Item{{name: "A", sellIn: 1, quality: 1}}, want: []*Item{{name: "A", sellIn: 0, quality: 0}}},
+		{testname: "A 1 0", in: []*Item{{name: "A", sellIn: 1, quality: 0}}, want: []*Item{{name: "A", sellIn: 0, quality: 0}}},
+		{testname: "Sulfuras, Hand of Ragnaros 0 1", in: []*Item{{name: "Sulfuras, Hand of Ragnaros", sellIn: 0, quality: 1}}, want: []*Item{{name: "Sulfuras, Hand of Ragnaros", sellIn: 0, quality: 1}}},
+		{testname: "Aged Brie 1 50", in: []*Item{{name: "Aged Brie", sellIn: 1, quality: 50}}, want: []*Item{{name: "Aged Brie", sellIn: 0, quality: 50}}},
+		{testname: "Aged Brie 1 49", in: []*Item{{name: "Aged Brie", sellIn: 1, quality: 49}}, want: []*Item{{name: "Aged Brie", sellIn: 0, quality: 50}}},
```

Comment:
If you have the same struct used in the tests again and again just export it to a variable